src/components/mask-input/mask-input-base.ts

```diff
@@ -0,0 +1,153 @@
+import { property, state } from 'lit/decorators.js';
+import { IgcInputBaseComponent } from '../input/input-base';
+import { MaskParser } from './mask-parser';
+interface MaskSelection {
+  start: number;
+  end: number;
+}
+
+export abstract class IgcMaskInputBaseComponent extends IgcInputBaseComponent {
+  protected parser = new MaskParser();
+  protected selection: MaskSelection = { start: 0, end: 0 };
+  protected compositionStart = 0;
+
+  @state()
+  protected hasFocus = false;
+
+  @state()
+  protected maskedValue = '';
+
+  /** The mask pattern to apply on the input. */
+  @property()
+  public mask!: string;
```

Comment:
Ahem, I thought the point of having the mask base was to reuse the functionality without inheriting all the public props of the mask that don't make sense to a date input.

In that regard - the `mask` doesn't fit. We should keep a protected backing prop in the base for functionality and only expose public get/set on the mask input